### ISSUE TITLE
Disable case-sensitiveness for future, newly created accounts

### ISSUE THREAD

**Issue by mensfeld** [rubygems/rubygems.org#3119]
## Is your feature request related to a problem?

Yes. At the moment, account names in RubyGems are case-sensitive. This means, that those two links direct to different profiles:

- https://rubygems.org/profiles/Mencio
- https://rubygems.org/profiles/mencio

or

- https://rubygems.org/profiles/shopify
- https://rubygems.org/profiles/Shopify (in case anyone from Shopify reads it, that's me - I will remove this account once this matter is solved in any way)

It is a security problem that can lead to a few potential problems (some now, some in the future):

1. Brandjacking - now we can create an account that has the same name (just a different letter size) for any account on RubyGems. This can lead to fake accounts being created in order to mimic the accounts for malicious reasons. 
2. In the future if we decide to introduce namespaces, this will be extremely problematic and bug/error-prone. `@rails` vs `@Rails`, etc. We will have to make sure that package resolution is also case sensitive due to namespaces being case sensitive.
3. Camel-case may lead to potential not-yet-discovered issues in the logic of handling packages and users, similar to how it happened with the platform and names collision.

I checked how it is done for NPM and NPM disallows the same names of a different case.

## Describe the solution you'd like

I would love us to:

1. Keep the current behavior for already existing accounts  OR migrate them to lowercase in case there are no conflicts
2. We need to assess how many accounts that are in use would be subject to collision and depending on scale, we would have to deal with that manually. Since at the moment we do not have namespaces, the collisions are not affecting package installs. This means, that aside from (potentially) profile URL changes, it should not have an extremely big impact.
4. We should redirect any camel-case entered profile to the downcased version (as long as it is not an exception if any)
5. We should disable the ability to use non-downcase letters in profile names.

## Describe alternatives you've considered

Alternative would be to keep it as it is

## Middle ground

The middle ground that would be fairly easy to implement and may be worth considering is *just* disabling the ability to use capitalized letters for new accounts. This should be fairly easy to implement and would not break anything in the system.

**Comment by sonalkr132** [comment]
> Camel-case may lead to potential not-yet-discovered issues in the logic of handling packages and users, similar to how it happened with the platform and names collision.

This is indeed a concern. At least with above PR any new account won't be possible.
We have 3339 existing accounts with this issue. 80 of these have over 1_000_000 total downloads. IMHO, we can just block the accounts which have lower downloads count for 1mil case.

**Comment by mensfeld** [comment]
I agree. Do you want me to close this issue since future cases are now handled?

**Comment by duckinator** [comment]
> We have 3339 existing accounts with this issue.

Do you mean accounts where the username isn't lowercase, or specifically accounts where there's two variants with different capitalization?

**Comment by sonalkr132** [comment]
> specifically accounts where there's two variants with different capitalization?

this.

> Do you want me to close this issue since future cases are now handled?

Perhaps you can add a todo to your issue and leave handinling for existing account unchecked.

**Comment by mensfeld** [comment]
@sonalkr132 I think we can safely close it since the potential of exploitation after your fix is much smaller. Now it would require getting an account that conflicts with a popular one and ATOing it and then finding a potential vulnerability. I will close it.

Thank you for a quick turnaround on this one :pray:
